crates/miden-testing/tests/integration/scripts/in_flight_swap.rs

```diff
@@ -0,0 +1,239 @@
+use miden_crypto::{Word, rand::FeltRng};
+use miden_lib::{
+    account::wallets::{AuxWallet, BasicWallet},
+    note::utils::{build_p2id_recipient, build_swap_tag},
+    transaction::TransactionKernel,
+};
+use miden_objects::{
+    Felt,
+    account::{Account, AccountId},
+    asset::Asset,
+    crypto::rand::RpoRandomCoin,
+    note::{
+        Note, NoteAssets, NoteDetails, NoteExecutionHint, NoteInputs, NoteMetadata, NoteRecipient,
+        NoteScript, NoteTag, NoteType,
+    },
+    transaction::OutputNote,
+};
+use miden_testing::{AccountState, Auth, MockChain};
+use rand::random;
+
+#[test]
+fn test_inflight_swap() {
+    let mut mock_chain = MockChain::new();
+    let offered_asset =
+        mock_chain.add_pending_new_faucet(Auth::BasicAuth, "NP", 100_000_u64).mint(200);
+    let requested_asset =
+        mock_chain.add_pending_new_faucet(Auth::BasicAuth, "MID", 100_000_u64).mint(100);
+
+    // For end users: Create a standard sender account with just BasicWallet
+    let mut alice = mock_chain.add_pending_existing_wallet(Auth::BasicAuth, vec![offered_asset]);
+    let mut bob = mock_chain.add_pending_existing_wallet(Auth::BasicAuth, vec![requested_asset]);
+
+    let (alice_note, alice_payback_note_details) =
+        create_in_flight_swap_note(alice.id(), offered_asset, requested_asset);
+
+    let (bob_note, bob_payback_note_details) =
+        create_in_flight_swap_note(bob.id(), requested_asset, offered_asset);
+
+    // For the matcher: Create the account with the AuxWallet as well as BasicWallet components
+    let account_builder =
+        Account::builder(random()).with_component(BasicWallet).with_component(AuxWallet);
+
+    let mut matcher_account = mock_chain.add_pending_account_from_builder(
+        Auth::BasicAuth,
+        account_builder,
+        AccountState::Exists,
+    );
+
+    let consume_swap_note_tx = mock_chain
+        .build_tx_context(matcher_account.id(), &[], &[alice_note, bob_note])
+        .build()
+        .execute()
+        .unwrap();
+
+    matcher_account.apply_delta(consume_swap_note_tx.account_delta()).unwrap();
+    // The matcher's vault should not have been affected
+    assert!(matcher_account.vault().assets().count() == 0);
```

Comment:
If we pass `Auth::NoAuth` to where we create the matcher account, then we should be able to make this assertion even stronger, i.e. the matcher account should not have been touched at all, so basically:

```rust
assert!(consume_swap_note_tx.account_delta().is_empty());
assert!(consume_swap_note_tx.account_delta().nonce().is_none());
```